tests/test_writer.py

```diff
@@ -105,81 +107,159 @@ def test_write_int_float_columns(self, tmp_path):
+        write_tfs(write_location, df, validate=None)  # don't care to validate
+        assert write_location.is_file()
+
+        written = write_location.read_text()
+        assert "PATH" in written
+        assert str(tmp_path) in written  # should be in its .__str__ form here
+
+    # ----- Below are tests for files with MAD-NG features ----- #
+
+    def test_tfs_write_read_with_booleans(self, _tfs_dataframe_booleans, tmp_path):
+        write_location = tmp_path / "test.tfs"
+        write_tfs(write_location, _tfs_dataframe_booleans, validate="madng")  # booleans are MAD-NG feature
+        assert write_location.is_file()
+
+        new = read_tfs(write_location)
+        assert_tfs_frame_equal(_tfs_dataframe_booleans, new, check_exact=False)  # float precision can be an issue
+
+    def test_tfs_write_read_with_complex(self, _tfs_dataframe_complex, tmp_path):
+        write_location = tmp_path / "test.tfs"
+        write_tfs(write_location, _tfs_dataframe_complex, validate="madng")  # booleans are MAD-NG feature
+        assert write_location.is_file()
+
+        new = read_tfs(write_location)
+        assert_tfs_frame_equal(_tfs_dataframe_complex, new, check_exact=False)  # float precision can be an issue
+
+    def test_tfs_write_with_nil_in_headers(self, _tfs_dataframe, tmp_path):
+        df = _tfs_dataframe
+        df.headers["WRITETONIL"] = None
+
+        write_location = tmp_path / "test.tfs"
+        write_tfs(write_location, df, validate="madng")
+
+        written = write_location.read_text()
+        assert "WRITETONIL" in written
+        assert "%n" in written
+        assert "nil" in written
+
+    def test_tfs_write_read_madng_like(self, _tfs_dataframe_madng, tmp_path):
+        write_location = tmp_path / "test.tfs"
+        write_tfs(write_location, _tfs_dataframe_madng, validate="madng")  # booleans are MAD-NG feature
+        assert write_location.is_file()
+
+        new = read_tfs(write_location)
+        assert_tfs_frame_equal(_tfs_dataframe_madng, new, check_exact=False)  # float precision can be an issue
+
+    @pytest.mark.skipif(sys.platform == "win32", reason="MAD-NG not available on Windows")
+    def test_tfs_write_madng_compatible_is_read_by_madng(self, _tfs_dataframe_madng, tmp_path):
+        from pymadng import MAD
+
+        write_location = tmp_path / "test.tfs"
+        write_tfs(write_location, _tfs_dataframe_madng, validate="madng")  # has all MAD-NG features
+        assert write_location.is_file()
+
+        # Now we read the file with pymadng and check there is not error from MADNG
+        # In case of error, pymadng will print the output to stdout/stderr. We need
+        # to ask to receive back from MAD-NG which would raise (and fail the test)
+        # if the loading did not go properly (our file is not accepted by MAD-NG).
+        madng = MAD(debug=True)
+        madng.protected_send(f"mtbl = mtable:read('{str(write_location.absolute())}')")
```

Comment:
I think with my changes in 0.6.0, this is equivalent.
```suggestion
        madng.send(f"mtbl = mtable:read('{str(write_location.absolute())}')")
```

Reply:
Indeed, fixed.